### PR TITLE
Feat/get dom

### PR DESCRIPTION
**Pull request by KyleMBoomer**
Going from empty files to this PR is quite the jump. I saved this PR for your inspection, so feel free to take your time reviewing: 
index.js
dom.js
api.js
index.html
styles.css

There is quite a bit going on already. If you have any questions, feel free to hit me up. Thank you for your review.

### CODE REVIEW

File: js/api.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+function fetchData(url) {
```
Comment (peterkimpk1) [inline js/api.js:1]: This fetch function can be more dynamic by having two parameters so it can be used for other types of network requests. It was mentioned in our code review but the URL parameter can be dynamic as well

File: js/dom.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+const loadData = () => {
+    return Promise.all([
+        fetchData('http://localhost:3001/api/v1/travelers/'),
+        fetchData('http://localhost:3001/api/v1/trips/'),
+        fetchData('http://localhost:3001/api/v1/destinations/')])
+}
+
+function displayUserData() {
+    const userID = parseInt(document.getElementById('userID').value)
```
Comment (peterkimpk1) [inline js/dom.js:21]: The value from userID is always a number so you can probably get rid of this line

File: js/dom.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+const displayTravelerData = (data) => {
+    if (data) {
+        travelerDetails.innerHTML = `
+            <p>Name: ${data.name}</p>
+            <p>Your Travel Penchant: ${data.travelerType}</p>
+        `
+    } else {
+        travelerDetails.innerHTML = `${data.error}`
```
Comment (peterkimpk1) [inline js/dom.js:49]: Since you're passing in fetch data, I don't think this else will invoke as your catch from fetchData function will show instead

File: js/dom.js
Comment (peterkimpk1) [inline js/dom.js:1]: Great uses of forEach to display items for all array items. Many of the if checks I think can be avoided since the user can't pass incorrect params through these functions and it's on you to decide what data type you're passing through in the params.

File: js/index.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    }
+    const { approvedTrips } = userTrips;
+    if (!approvedTrips.length) {
+        return 'All your trips are pending approval.';
+    }
+
+    const expendituresByYear = {};
+
+    approvedTrips.forEach(trip => {
+        const tripYear = new Date(trip.date).getFullYear();
```
Comment (peterkimpk1) [inline js/index.js:33]: I like that you're able to see expenditures by different years but since the project scope asks for the latest user year, I'd try to highlight that in the function itself. You can do a separate forEach to get the highest number year

File: test/test.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    const { pendingTrips } = getUserTrips(4, trips)
+    const pendingTrip = [
+      { date: "2023/08/05", destinationID: 8, duration: 6, id: 5, status: "pending", suggestedActivities: ["diving"], travelers: 1, userID: 4 }
+    ]
+    expect(pendingTrips).to.deep.equal(pendingTrip)
+  })
+
+  describe('Get User Expenditures', function () {
+    it('should return total money spent on all approved trips in the target year, plus 10%, based on UserID', () => {
+      const totalBill = getUserExpenditures(4, trips, destinations, 2023)
```
Comment (peterkimpk1) [inline test/test.js:47]: This test function with getUserExpenditures has 4 params, but I'm only seeing 3 params in the same function in index.js

File: test/test.js
Comment (peterkimpk1) [inline test/test.js:1]: Great use of before to clean up the repetitiveness. However you refactor the getUserExpenditures function, I'd like to see a test that's able to return the user's latest year's expenditures without passing a year parameter.

Comment (peterkimpk1) [inline test/test.js:1]: We mentioned briefly about sad paths in our live code review, but always considering too many edge cases for each function will slow down development, and that's something to consider going forward to these next few iterations. Also, I'd make sure to update the tests first before making changes to user functions

File: dist/index.html
Comment (peterkimpk1) [inline dist/index.html:1]: Something to note here is the many uses of IDs. Since classes offer reusability with CSS, I'd like to see more utilization of classes as you develop this HTML so declaring CSS rules will be easier. I think generally classes should be used first, and IDs should be reached for second for uniqueness. Since the HTML elements have ids declared first, and classes second, or vice versa, I'd like to see some unison in the declarations.

**Comment by peterkimpk1** [comment]
Overall, this PR contained a lot of changes for 1 PR. I'd try to commit a little more frequently, and push up feature changes with smaller PRs to main.
